### PR TITLE
history bugfix

### PR DESCRIPTION
**Pull request by llllllllll**

### CODE REVIEW

File: zipline/history/history.py [hunk 1]
```diff
@@ -278,6 +285,8 @@ def __init__(self, bar_count, frequency, field, ffill,
         # The frequency at which the data is sampled.
         self.frequency = frequency
         # The field, e.g. 'price', 'volume', etc.
+        if not isinstance(field, Hashable):
```
Comment (ssanderson) [inline zipline/history/history.py]: Not sure if I have a preference if they're a equivalent, but is this functionally different from:
```
try:
    hash(field)
except:
    raise ValueError(...)
```
?

File: zipline/history/history.py [hunk 2]
```diff
@@ -278,6 +285,8 @@ def __init__(self, bar_count, frequency, field, ffill,
...
+            raise ValueError('field must be hashable')
```
Comment (ssanderson) [inline zipline/history/history.py]: This should reference what the actual received field was.

File: zipline/history/history.py [hunk 3]
```diff
@@ -266,6 +270,9 @@ def __init__(self, bar_count, frequency, field, ffill,
                  data_frequency='daily'):
 
         # Number of bars to look back.
+        if not isinstance(bar_count, int):
+            raise ValueError('bar_count must be an int')
```
Comment (ssanderson) [inline zipline/history/history.py]: This should reference the actual received type so the user can do something about it.

File: zipline/history/history.py (hunk 1 above)
Comment (llllllllll) [inline zipline/history/history.py]: This is only difference if you are a bully and do something like:

```
class C(object):
    @property
    def __hash__(self):
        raise AttributeError()
```

But the abc is cleaner imho

Comment (ssanderson) [inline zipline/history/history.py]: So the abc is equivalent to `hasattr('__hash__')`?

Comment (llllllllll) [inline zipline/history/history.py]: err no, that is `hasattr(type(c))`

**Comment by llllllllll** [comment]
@ssanderson looks good?

File: zipline/history/history.py [hunk 4]
```diff
@@ -266,6 +270,14 @@ def __init__(self, bar_count, frequency, field, ffill,
                  data_frequency='daily'):
 
         # Number of bars to look back.
+        if not isinstance(bar_count, int):
+            raise ValueError(
```
Comment (ssanderson) [inline zipline/history/history.py]: I think this should be a TypeError

File: zipline/history/history.py (hunk 1 above)
Comment (ssanderson) [inline zipline/history/history.py]: Looking at this again, I think we should just assert that this is in HistoryContainer.VALID_FIELDS.  We might have to move that somewhere shared to fix a circular dependency.

Comment (ssanderson) [inline zipline/history/history.py]: (Where "assert" might mean "check and raise a ValueError".)

**Comment by jbredeche** [comment]
@llllllllll this is likely no longer needed due to the recent history rewrite, what do you think?

**Pull request closed without merging**